tests/Http/Three/SignupTest.php

```diff
@@ -281,10 +281,60 @@ public function testUnauthenticatedUserUpdatingASignup()
         $signup = factory(Signup::class)->create();
 
         $response = $this->patchJson('api/v3/signups/' . $signup->id, [
-            'quantity'     => 888,
             'why_participated'  => 'new why participated',
         ]);
 
         $response->assertStatus(401);
     }
+
+    /**
+     * Test to make sure we are returning quantity correctly.
+     * Quantity is either summed total of quantity across all posts under a signup
+     * or quanttiy under signup if it is still on signup record.
+     *
+     * GET /signups
+     * @return void
+     */
+    public function testQuantityOnSignupIndex()
+    {
+        // Create one signup with a quantity_pending.
+        $firstSignup = factory(Signup::class)->create();
+
+        // Create another signup with a quantity.
+        $secondSignup = factory(Signup::class)->create();
+        $secondSignup->quantity_pending = null;
+        $secondSignup->quantity = 8;
+        $secondSignup->save();
+
+        // Create another signup with three posts with quantities.
+        $thirdSignup = factory(Signup::class)->create();
+        $thirdSignup->quantity_pending = null;
+        $thirdSignup->save();
+
+        for ($i = 0; $i < 3; $i++) {
```

Comment:
If you update the `PostFactory` in `ModelFactory.php` so that it creates a `Post` with a random `quantity`, you can then create three posts with quantities with ` $posts = factory(Post::class, 3)->create();` 

Later in your test you can sum up the quantities over the set of posts you created, and confirm that the quantity you received is equal to the quantity you summed up. 

I think that would be cool, because then we can pretty easily create `posts` with quantities in our tests whenever we need them.

Reply:
Great idea! 💡

Reply:
updated this and also `testQuantityOnActivityIndex` to use the same logic with the updated `PostFactory`!